test/test-form.js

```diff
@@ -379,6 +431,35 @@ test('handle error', function (t) {
     t.end();
 });
 
+test('promised handle error', function (t) {
+    // eslint-disable-next-line no-undef
+    if (typeof Promise === 'function') {
+        return t.end();
+    }
+    t.plan(6);
+    var f = forms.create({ field1: forms.fields.string() });
+    f.bind = function () {
+        t.ok(true, 'bind called');
+        f.fields.field1.error = 'some error';
+        f.isValid = function () { return false; };
+        return f;
+    };
+    f.validate = function (callback) {
+        t.ok(true, 'validate called');
+        callback(null, f);
+    };
+    return f.handle({ field1: 'test' }).then(function (result) {
+        t.ok(result.error, 'handle promise resolved to error');
+        t.notOk(result.empty, 'handle promise correctly did not resolve to empty');
+        t.notOk(result.success, 'handle promise correctly did not resolve to success');
+        t.equal(result.form, f);
+        t.end();
+    }, function () {
+        t.fail('handle promise was rejected');
+        t.end();
```

Comment:
no need to call `t.end()` when the test returns a promise.

Reply:
Cool, I'm a bit of a newbie with `tape`, usually sits with Mocha, thanks 👍